### PR TITLE
Confirmation before delete download

### LINKED ISSUE #5419: Confirmation before deleting all downloads
### Checklist

- [x] I checked, but didn't find any duplicates (open OR closed) of this issue in the repo. 
- [ x] I have read the contribution guidelines given at https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md.
- [ ] This issue contains only one feature request. I will open one issue for every feature I want to request.

My suggestion is to make a pop up confirmation when selecting the "delete downloaded files" option or a trash can so we can recover those deleted files

Having a single touch wipe is way too dangerous for a backup
As my own experience, a single missclick cost all my backup of temporary videos I had for studying

### PR DESCRIPTION
**Pull request by Marc2Serre**
#### What is it?
- [ ] Bugfix (user facing)
- [x] Feature (user facing)
- [ ] Codebase improvement (dev facing)
- [x] Meta improvement to the project (dev facing)

#### Description of the changes in your PR
We have a confirmation dialog when we delete a single file from the download menu like in the screenshot below. We didn't translate the text. It is only in english. How the translation works? Is done automatically or manually?

#### Before/After Screenshots/Screen Record

- After:
image

#### Fixes the following issue(s)

- Fixes #5419 

#### APK testing 

The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR.

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

File: app/src/main/java/us/shandian/giga/ui/adapter/MissionAdapter.java [hunk 1]
```diff
@@ -678,9 +678,19 @@ private boolean handlePopupItem(@NonNull ViewHolderItem h, @NonNull MenuItem opt
...
+                            mDeleter.append(h.item.mission);
+                            applyChanges();
+                            checkMasterButtonsVisibility();
+                        })
+                        .setNegativeButton(R.string.cancel, (dialog, which) -> {
+                            //nothing happens
+                        });
+                // Create the AlertDialog object and run it
+                builder.create();
+                builder.show();
```
Comment (Stypox) [inline app/src/main/java/us/shandian/giga/ui/adapter/MissionAdapter.java:693]: This improves the formatting, removes an unneeded variable, shows the name of the file being deleted in the dialog.
```suggestion
                new AlertDialog.Builder(mContext)
                        .setMessage(mContext.getString(R.string.delete_selected_file, h.item.mission.storage.getName()))
                        .setNegativeButton(R.string.cancel, null)
                        .setPositiveButton(R.string.yes, (dialog, which) -> {
                            mDeleter.append(h.item.mission);
                            applyChanges();
                            checkMasterButtonsVisibility();
                        })
                        .show();
```

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -583,6 +583,7 @@
     <string name="clear_download_history">Clear download history</string>
     <string name="confirm_prompt">Do you want to clear your download history or delete all downloaded files?</string>
     <string name="delete_downloaded_files">Delete downloaded files</string>
+    <string name="delete_selected_file">Do you want to delete the selected file ?</string>
```
Comment (Stypox) [inline app/src/main/res/values/strings.xml:586]: This shows the file being deleted in the string
```suggestion
    <string name="delete_selected_file">Do you want to delete %1$s?</string>
```

**Review by Stypox — CHANGES_REQUESTED** [review]
Thank you :-)

> We didn't translate the text. It is only in english.

That's ok. It will be translated on Weblate separately from this PR, see the explanation in the README and in the contribution guidelines.

**Review by TiA4f8R — CHANGES_REQUESTED** [review]
Thank you for your pull request!

> We didn't translate the text. It is only in english. How the translation works? Is done automatically or manually?

The translation is made by translators on [our Weblate](https://hosted.weblate.org/projects/newpipe/), but you can add translations of the languages you are speaking in your PR if you want (in the `strings.xml` resource file, which corresponds to language set in the folder: for example, the `strings.xml` file of the `values-fr` folder contains the French translations of the app).
If a translation is not available, string resources of the original language will be displayed (English in our case).

You need to do some changes first and then your changes should be good to merged.

Question for other team members: should we remove the toast displayed to cancel the deletion after the AlertDialog is dismissed in the case the deletion is confirmed by the user or not?

**Comment by Stypox** [comment]
@TiA4f8R I think @Marc2Serre would be confused to see two reviews asking him to do two different things ;-)

> Question for other team members: should we remove the toast displayed to cancel the deletion after the AlertDialog is dismissed in the case the deletion is confirmed by the user or not?

What? ;-)

**Comment by Hugov8** [comment]
Thanks for the feedback, we change the code with your suggestions and we add the french translation

**Comment by litetex** [comment]
Question to the devs: Do we really need this in the first place?

There are already multiple fail-safes so I don't think we need another one:
* When deleting a single file a snackbar shows up (at the bottom) which allows undoing the deletion
* When clicking at the trash icon in the top right corner a dialog shows up which asks you if you really want to delete all downloaded files

So what's the point of having this? I see no advantage.

**Comment by litetex** [comment]
This PR will also not solve #5419 "Confirmation before deleting >**all**< downloads".
The issue is about the trash can in the top right corner, not about the deletion of a single file.
I think we should unlink the PR and the issue.

**Comment by Stypox** [comment]
> So what's the point of having this? I see no advantage.

Mmmh, didn't think about this, I think you are right

**Comment by litetex** [comment]
Killing this PR as it doesn't introduces a (noteable) benefit and doesn't fix the underlying issue:
* https://github.com/TeamNewPipe/NewPipe/pull/7435#issuecomment-979464947
* https://github.com/TeamNewPipe/NewPipe/pull/7435#issuecomment-979467969

**Pull request closed without merging**